### PR TITLE
GH-40216: [CI][Packaging][Python] Upload pyarrow nightly wheels to scientific python channel on Anaconda

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

As discussed on the main issue is interesting for discoverability to have the wheels uploaded to the nightly channel.

### What changes are included in this PR?

Added macro to upload wheel to scientific python channel

### Are these changes tested?

Via archery

### Are there any user-facing changes?

No but nightly wheels will be available on scientific python channel
* GitHub Issue: #40216

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
we can't use the action on non-linux. I'll try to upload manually

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
Revision: 8662076458fcb583f995b48f4ece560480d774b9

Submitted crossbow builds: [ursacomputing/crossbow @ test-upload-scientific-2](https://github.com/ursacomputing/crossbow/branches/all?query=test-upload-scientific-2)

|Task|Status|
|----|------|
|python-sdist|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216678/job/29416568365)|
|wheel-macos-big-sur-cp310-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216633/job/29416567731)|
|wheel-macos-big-sur-cp311-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217104/job/29416570692)|
|wheel-macos-big-sur-cp312-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216936/job/29416569711)|
|wheel-macos-big-sur-cp313-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217302/job/29416572041)|
|wheel-macos-big-sur-cp38-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216553/job/29416567324)|
|wheel-macos-big-sur-cp39-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216974/job/29416569942)|
|wheel-macos-catalina-cp310-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216611/job/29416567816)|
|wheel-macos-catalina-cp311-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216987/job/29416570056)|
|wheel-macos-catalina-cp312-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217504/job/29416573002)|
|wheel-macos-catalina-cp313-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216605/job/29416567595)|
|wheel-macos-catalina-cp38-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217161/job/29416570913)|
|wheel-macos-catalina-cp39-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216812/job/29416568884)|
|wheel-manylinux-2-28-cp310-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217331/job/29416572065)|
|wheel-manylinux-2-28-cp310-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217045/job/29416570345)|
|wheel-manylinux-2-28-cp311-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216984/job/29416569943)|
|wheel-manylinux-2-28-cp311-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217176/job/29416571293)|
|wheel-manylinux-2-28-cp312-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217230/job/29416571163)|
|wheel-manylinux-2-28-cp312-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217348/job/29416571979)|
|wheel-manylinux-2-28-cp313-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216790/job/29416568324)|
|wheel-manylinux-2-28-cp313-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217218/job/29416571427)|
|wheel-manylinux-2-28-cp38-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216908/job/29416569432)|
|wheel-manylinux-2-28-cp38-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217307/job/29416571786)|
|wheel-manylinux-2-28-cp39-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217267/job/29416571721)|
|wheel-manylinux-2-28-cp39-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216925/job/29416569431)|
|wheel-manylinux-2014-cp310-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216912/job/29416569349)|
|wheel-manylinux-2014-cp310-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216878/job/29416569047)|
|wheel-manylinux-2014-cp311-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217148/job/29416570872)|
|wheel-manylinux-2014-cp311-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217239/job/29416571171)|
|wheel-manylinux-2014-cp312-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216771/job/29416568294)|
|wheel-manylinux-2014-cp312-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216890/job/29416569182)|
|wheel-manylinux-2014-cp313-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216765/job/29416568568)|
|wheel-manylinux-2014-cp313-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217509/job/29416573114)|
|wheel-manylinux-2014-cp38-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216830/job/29416568953)|
|wheel-manylinux-2014-cp38-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217177/job/29416571200)|
|wheel-manylinux-2014-cp39-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217270/job/29416571706)|
|wheel-manylinux-2014-cp39-arm64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216910/job/29416569062)|
|wheel-windows-cp310-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217028/job/29416570362)|
|wheel-windows-cp311-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216625/job/29416567583)|
|wheel-windows-cp312-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217388/job/29416572341)|
|wheel-windows-cp313-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217500/job/29416572826)|
|wheel-windows-cp38-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613216535/job/29416567323)|
|wheel-windows-cp39-amd64|[GitHub Actions](https://github.com/ursacomputing/crossbow/actions/runs/10613217089/job/29416570679)|

**Comment by raulcd** [comment]
@jorisvandenbossche all wheels have been uploaded successfully, see: https://anaconda.org/scientific-python-nightly-wheels/pyarrow/files

I will remove the version: `18.0.0.dev236` once we merge the PR and we can let the nightlies to get uploaded.

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ groups:
 
   wheel:
     - wheel-*
+    - python-sdist
```
Comment (raulcd) [inline dev/tasks/tasks.yml:32]: I've added the `sdist` as part of the wheels job as several times we want to also trigger it when we trigger the wheels group but happy to remove it from here if we want to have it separate

**Review by kou — APPROVED** [review]
+1

Can we drop support for Gemfury?

**Review by assignUser — APPROVED** [review]
Nice, much easier than doing it here!

>  Can we drop support for Gemfury?

I think we should discuss that in a separate issue?

**Comment by raulcd** [comment]
I think is a good idea to drop uploading wheels to gemfury. I've created:
- https://github.com/apache/arrow/issues/43904

to follow that up.

**Pull request merged**